.github/workflows/update-grpc.yml

```diff
@@ -0,0 +1,74 @@
+name: Update gRPC source
+
+on:
+  workflow_dispatch:
+    inputs:
+      grpc-common-tag:
+        type: string
+        description: "tag name of th2-grpc-common repository"
+
+
+env:
+  MAKEFILE: Makefile
+  VERSION_VARIABLE: TH2_GRPC_COMMON_VERSION
+
+jobs:
+  update-grpc:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: 'Checkout repository'
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Get nearest tag
+        run: echo "NEAREST_TAG=$(git describe --abbrev=0 --tags)" >> $GITHUB_ENV
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - name: Install protoc
+        run: make install-protoc
+      - name: Check gRPC generated files not manually changed
+        shell: bash
+        run: make generate-grpc-files
+      - name: Update th2-grpc-common tag in Makefile
+        shell: bash
+        run: >
+          sed --in-place "s/^${{ env.VERSION_VARIABLE }}=.*/${{ env.VERSION_VARIABLE }}=${{ inputs.grpc-common-tag }} # Updated by Github workflow/" "${{ env.MAKEFILE }}"
+      - name: Install protoc
+        run: make install-protoc
+      - name: Check gRPC generated files not manually changed
+        shell: bash
+        run: make check-grpc-files
+      - name: Check sources
+        run: make run-test
+      - name: Build sources
+        run: make build
+      - name: Commit changes and create tag
+        uses: EndBug/add-and-commit@v9
+        with:
+          add: "['*.go', '${{ env.MAKEFILE }}']"
+          default_author: github_actions
+          message: "Updated gRPC source using th2-grpc-common:${{ inputs.grpc-common-tag }}"
+          tag: ${{ inputs.grpc-common-tag }}
+      - name: Build changelog
+        id: build_changelog
+        uses: mikepenz/release-changelog-builder-action@v5
+        with:
+          mode: "COMMIT"
+          configuration: .github/workflows/update-grpc-changelog-config.json
+          fromTag: ${{ env.NEAREST_TAG }}
+          toTag: ${{ inputs.grpc-common-tag }}
```

Comment:
question: do we need to collect the change log? The only thing it will contain is the updates of the version for proto files

Reply:
I think yes just in case other commits